packages/additionalapi/src/database/ImportLogic.ts

```diff
@@ -204,27 +206,26 @@ function prepareInputStreamContainmentsFlatBuffers(bulkImport: FBBulkImport, met
 }
 
 async function pipeInputIntoQueryStream(client: PoolClient, query: string, inputStream: Duplex, opDesc: string): Promise<void> {
-    await new Promise<void>((resolve, reject) => {
-        try {
-            const queryStream = client.query(copyFrom(query))
-
-            inputStream.on("error", (err: Error) => {
-                reject(`Input stream error on ${opDesc}: ${err}`)
-            })
+    const queryStream = client.query(copyFrom(query));
 
-            queryStream.on("error", (err: Error) => {
-                reject(`Query stream error on ${opDesc}: ${err}`)
-            })
+    // propagate src errors too
+    const srcErr = new Promise<never>((_, reject) =>
+        inputStream.once("error", (e) =>
+            reject(new Error(`Error on ${opDesc} (source): ${e instanceof Error ? e.message : String(e)}`))
+        )
+    );
 
-            inputStream.on("end", () => {
-                resolve()
-            })
+    inputStream.pipe(queryStream);
 
-            inputStream.pipe(queryStream)
-        } catch (e) {
-            reject(`Error on ${opDesc}: ${e}`)
-        }
-    })
+    try {
+        // COPY FROM completes on the writable's 'finish'
+        await Promise.race([
+            finished(queryStream), // resolves on finish, rejects on error/close
+            srcErr,
+        ]);
+    } catch (e) {
+        throw new Error(`Error on ${opDesc}: ${e instanceof Error ? e.message : String(e)}`);
```

Comment:
Is it intentional to throw an `Error` here, instead of `Promise.reject(...)`?

Reply:
I added the test (Verify what happens when throwing in async)